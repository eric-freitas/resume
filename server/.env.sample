PORT="8094"
SECRET_TOKEN="xxxx"
GOOGLE_CLIENTID="xxxxxxxxxxxxxx"
GOOGLE_SECRET="xxxxxxxxxxx"
PG_CONN="{"host":"motty.db.elephantsql.com","port":5432,"database":"DB","schema":"SCHEMA","user":"USER","password":"PASS"}"
ELASTIC_CONN="{"node":"https://rqz1csj7xg:hhba2m2e4i@duval-dev-513539397.us-east-1.bonsaisearch.net:443","auth":{"username":"USER","password":"PASS"}}"
MONGO_CONN="mongodb+srv://USER:PASS@duval.bdatb.mongodb.net/DB?retryWrites=true&w=majority"