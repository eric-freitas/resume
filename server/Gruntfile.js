const pkgJson = require("./package.json");

module.exports = (grunt) => {
    const getConfig = () => {
        return {
            copy: {
                main: {
                    files: [
                        {
                            cwd: "src",
                            src: "**",
                            dest: "build",
                            expand: true
                        },
                        {
                            cwd: "nssm",
                            src: "**",
                            dest: "build",
                            expand: true
                        },
                        {
                            cwd: "",
                            src: ["package.json", "README.md", "CHANGELOG.md", "*.bat"],
                            dest: "build",
                            expand: true
                        }
                    ]
                }
            },
            clean: ["build", "dist"],
            compress: {
                main: {
                    options: {
                        archive: `dist/build_v${pkgJson.version}.zip`
                    },
                    files: [{
                        expand: true,
                        cwd: "build/",
                        src: ["**"],
                        dest: "/"
                    }]
                }
            },
            secret: grunt.file.readJSON("secret.json"),
            environments: {}
        };
    };

    const setDeployEnvironments = (config) => {
        const _config = Object.assign({}, config);
        const secret = _config.secret;
        const secretKeys = Object.keys(secret);
        const deployEnvironments = {
            options: {
                local_path: "dist",
                current_symlink: "current",
                readyTimeout: 40000
            }
        };
        secretKeys.forEach((sk) => {
            deployEnvironments[sk] = {
                options: {
                    host: `<%= secret.${sk}.host %>`,
                    username: `<%= secret.${sk}.username %>`,
                    password: `<%= secret.${sk}.password %>`,
                    deploy_path: "/home/eric/applications/condomilson/releases",
                    releases_to_keep: "10",
                    after_deploy: `cd /home/eric/applications/condomilson/releases/current && unzip build_v${pkgJson.version}.zip && rm build_v${pkgJson.version}.zip && pm2 reload condomilson`,
                    debug: true
                }
            };
        });
        _config.environments = deployEnvironments;
        return _config;
    };

    grunt.initConfig(setDeployEnvironments(getConfig()));

    grunt.loadNpmTasks("grunt-contrib-copy");
    grunt.loadNpmTasks("grunt-contrib-clean");
    grunt.loadNpmTasks("grunt-contrib-compress");
    //grunt.loadNpmTasks("grunt-ssh-deploy");

    const registerGruntDynamicTasks = (grunt, config) => {
        const _config = Object.assign({}, config);
        const secret = _config.secret;
        const secretKeys = Object.keys(secret);
        secretKeys.forEach((sk) => {
            //grunt.registerTask(`deploy-${sk}`, [`ssh_deploy:${sk}`]);
            //grunt.registerTask(`rollback-${sk}`, [`ssh_rollback:${sk}`]);
            grunt.registerTask(`deploy-${sk}`, [`deploy:${sk}`]);
        });
    }

    const registerGruntTasks = (grunt, config) => {
        registerGruntDynamicTasks(grunt, config);
        grunt.registerTask("build", ["clean", "copy", "npm-install", "npm-audit-fix", "compress:main"]);

        grunt.registerTask("npm-install", "install the backend dependencies only", function () {
            var exec = require("child_process").exec;
            var cb = this.async();
            exec("npm install --only=production", { cwd: "./build" }, function (err, stdout, stderr) { // eslint-disable-line no-unused-vars
                cb(err);
            });
        });
    
        grunt.registerTask("npm-audit-fix", "apply npm audit fix", function () {
            var exec = require("child_process").exec;
            var cb = this.async();
            exec("npm audit fix", { cwd: "./build" }, function (err, stdout, stderr) { // eslint-disable-line no-unused-vars
                cb(err);
            });
        });
    };

    registerGruntTasks(grunt, getConfig());
};
