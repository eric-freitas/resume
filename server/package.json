{
  "name": "resume",
  "version": "1.0.0",
  "description": "Resume Server",
  "main": "index.ts",
  "scripts": {
    "dev": " npx ts-node-dev ./src/index.ts",
    "test": "jest",
    "build": "rimraf ./build && tsc",
    "start": "npm run build && node build/index.js",
    "coverage": "jest --coverage"
  },
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "@types/cookie-parser": "^1.4.2",
    "@types/cors": "^2.8.6",
    "@types/dotenv-flow": "^3.0.0",
    "@types/express": "^4.17.6",
    "@types/helmet": "0.0.47",
    "@types/jest": "^26.0.0",
    "@types/jsonwebtoken": "^8.5.0",
    "@types/morgan": "^1.9.0",
    "@types/node": "^14.0.11",
    "@types/passport": "^1.0.3",
    "@types/passport-strategy": "^0.2.35",
    "@types/supertest": "^2.0.9",
    "babel-cli": "^6.26.0",
    "babel-preset-env": "^1.7.0",
    "jest": "^26.0.1",
    "jest-extended": "^0.11.5",
    "jest-junit": "^10.0.0",
    "nodemon": "^2.0.4",
    "rimraf": "^3.0.2",
    "superagent": "^5.2.2",
    "supertest": "^4.0.2",
    "ts-jest": "^26.1.0",
    "ts-node": "^8.10.2",
    "typescript": "^4.0.5",
    "ts-node-dev": "^1.0.0"
  },
  "dependencies": {
    "cls-hooked": "^4.2.2",
    "cors": "^2.8.5",
    "dotenv": "^8.2.0",
    "dotenv-flow": "^3.2.0",
    "express": "^4.17.1",
    "logform": "^2.2.0",
    "morgan": "^1.10.0",
    "pg": "^8.6.0",
    "request": "^2.88.2",
    "sniffr": "^1.2.0",
    "winston": "^3.3.3"
  }
}
