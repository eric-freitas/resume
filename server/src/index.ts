import express from 'express';
import routes from './routes';
import cors from 'cors';

import logger from 'morgan';

import dotenv from 'dotenv-flow';

import config from './utils/config';

dotenv.config();

const port = config.port;

const app = express();
const corsOption = {
    origin: true,
    methods: 'GET,HEAD,PUT,PATCH,POST,DELETE',
    credentials: true,
    exposedHeaders: ['x-auth-token']
};
app.use(cors(corsOption));
app.use(express.json());

app.use(logger('dev'));

app.use(routes);

const server = app.listen(port);
console.log("server open at port " + port);

export default server;