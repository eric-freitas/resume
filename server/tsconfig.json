{
  "compilerOptions": {
    "target": "es5",                          
    "module": "commonjs",                     
    "strict": true,                           
    "esModuleInterop": true,                   
    "skipLibCheck": true,                     
    "forceConsistentCasingInFileNames": true ,
    "resolveJsonModule": true,
    "noImplicitAny": false
  },
  "exclude": ["node_modules", "**/*.spec.ts", "**/*.test.ts"],
  "include": ["./src/**/*.tsx", "./src/**/*.ts"]
}
