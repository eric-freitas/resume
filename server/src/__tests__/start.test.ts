import 'jest-extended';
import request from 'supertest';
import server from '../index';

let app:any = server;

describe ("resume test", () => {

    it ('should access "/"', (done) => {
        request(app)
            .get("/")
            .then(response => {
            expect((response as any).statusCode).toBe(200);
            done();
        });
    })

	it ('should access "/test"', (done) => {
        
        request(app)
            .get("/test/1")
            .then(response => {
                expect((response as any).statusCode).toBe(200);
                const _d = response.body;
                expect(_d).not.toBeNull()
                expect(_d).toBeObject();
                const { id } = _d;
                expect(id).toBe(1);
            done();
        });
    })

    it ('should list "/skills"', (done) => {
        
        request(app)
            .get("/skills/pt")
            .then(response => {
                expect((response as any).statusCode).toBe(200);
                const _d = response.body;
                expect(_d).not.toBeNull()
                expect(_d).toBeArray()
                expect(_d).not.toBeArrayOfSize(0)

                _d.forEach(element => {
                    expect(element).toBeObject();
                    expect(element).not.toBeNull();

                    expect(element).toMatchSnapshot({
                        name        : expect.any(String),
                        level       : expect.any(Number),
                        maxLevel    : expect.any(Number)
                    })

                    expect(element.level).toBeLessThanOrEqual(element.maxLevel);
                    expect(element.name).not.toBeEmpty();
                });
            done();
        });
    })

});
  