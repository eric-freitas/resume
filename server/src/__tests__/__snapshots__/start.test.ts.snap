// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resume test should list "/skills" 1`] = `
Object {
  "level": Any<Number>,
  "maxLevel": Any<Number>,
  "name": Any<String>,
}
`;

exports[`resume test should list "/skills" 2`] = `
Object {
  "level": Any<Number>,
  "maxLevel": Any<Number>,
  "name": Any<String>,
  "text": "X",
}
`;
